polar_route/vessel_performance/vessels/abstract_plane.py

```diff
@@ -0,0 +1,138 @@
+from polar_route.vessel_performance.abstract_vessel import AbstractVessel
+from meshiphi.mesh_generation.aggregated_cellbox import AggregatedCellBox
+from abc import abstractmethod
+import logging
+
+
+class AbstractPlane(AbstractVessel):
+    """
+        Abstract class to model the performance of an underwater glider
+    """
+    def __init__(self, params):
+        """
+            Args:
+                params (dict): vessel parameters from the vessel config file
+        """
+        self.vessel_params = params
+        logging.info(f"Initialising a vessel object of type: {self.vessel_params['vessel_type']}")
+        self.max_speed      = self.vessel_params['max_speed']
+        self.speed_unit     = self.vessel_params['unit']
+        self.max_elevation  = -1 * self.vessel_params['min_depth']
+        self.max_ice        = self.vessel_params['max_ice_conc']
+        self.excluded_zones = self.vessel_params.get('excluded_zones')
+
+
+    def model_performance(self, cellbox):
+        """
+            Method to determine the performance characteristics for the underwater glider
+
+            Args:
+                    cellbox (AggregatedCellBox): input cell from environmental mesh
+        """
+        logging.debug(
+            f"Modelling performance in cell {cellbox.id} for a vessel of type: {self.vessel_params['vessel_type']}")
+        perf_cellbox = self.model_speed(cellbox)
+        perf_cellbox = self.model_fuel(perf_cellbox)
+
+        performance_values = {k: v for k, v in perf_cellbox.agg_data.items() if k not in cellbox.agg_data}
+
+        return performance_values
+
+    def model_accessibility(self, cellbox):
+        """
+            Method to determine if a given cell is accessible to the underwater glider
+
+            Args:
+                cellbox (AggregatedCellBox): input cell from environmental mesh
+            Returns:
+                access_values (dict): boolean values for the modelled accessibility criteria
+        """
+        logging.debug(f"Modelling accessibility in cell {cellbox.id} for a vessel of type: "
+                      f"{self.vessel_params['vessel_type']}")
+        access_values = dict()
+
+        # Exclude cells due to land or ice
+        
+        access_values['shallow'] = self.shallow(cellbox)
+        access_values['ext_ice'] = self.extreme_ice(cellbox)
+
+        # Exclude any other cell types specified in config
+        if self.excluded_zones is not None:
+            for zone in self.excluded_zones:
+                access_values[zone] = cellbox.agg_data[zone]
+
+        access_values['inaccessible'] = any(access_values.values())
+
+        access_values['land']    = self.land(cellbox)
```

Comment:
As I understand it, this will mark all land as inaccessible to any plane which isn't what you would want surely?

Reply:
You're right, my mistake :+1: